src/hooks/http.ts

```diff
@@ -13,13 +15,77 @@ export default function httpHook(mod: HTTP) {
       return target.apply(thisArg, argArray).prependListener("request", handleRequest);
     },
   });
+
+  hookClientRequestApi(mod);
+
   return mod;
 }
 
 httpHook.applicable = function (id: string) {
   return ["http", "https", "node:http", "node:https"].includes(id);
 };
 
+function hookClientRequestApi(mod: HTTP) {
+  const createApplyProxy = (f: HTTP["get" | "request"]) => {
+    return new Proxy(f, {
+      apply(target, thisArg, argArray: Parameters<typeof f>) {
+        const clientRequest = target.apply(thisArg, argArray);
+        handleClientRequest(clientRequest);
+        return clientRequest;
+      },
+    });
+  };
+
+  mod.request = createApplyProxy(mod.request);
+  mod.get = createApplyProxy(mod.get);
+
+  if ("ClientRequest" in mod) {
+    mod.ClientRequest = new Proxy(mod.ClientRequest, {
+      construct(target, argArray, newTarget): object {
+        const clientRequest = Reflect.construct(target, argArray, newTarget) as ClientRequest;
+        handleClientRequest(clientRequest);
+        return clientRequest;
+      },
+    });
+  }
+}
+
+const clientRequests = new WeakSet<http.ClientRequest>();
+
+function handleClientRequest(request: http.ClientRequest) {
+  // for some reason proxy construct/apply is called multiple times for the same request
+  // so let's make sure we haven't seen this one before
+  if (clientRequests.has(request)) return;
+  clientRequests.add(request);
+
+  const url = new URL(`${request.protocol}//${request.host}${request.path}`);
```

Comment:
Does `host` include the port number? If not, make sure to add it if it's not the default for the protocol.